SPECS/tdnf/tdnf-default-mariner-release.patch

```diff
@@ -1,24 +1,30 @@
-From: Andrew Phelps <anphel@microsoft.com>
-Date: Mon, 25 Nov 2019 04:16:16 +0000
-Subject: [PATCH] Merged PR 253: fix default distroverpkg
+From: Sam Meluch <sammeluch@microsoft.com>
+Date: Tue, 11 Apr 2023 17:44:54 +0000
+Subject: [PATCH] Fix the default distroverpkg
 
-fix default distroverpkg to prevent error:
-Error(1022) : distroverpkg config entry is set to a package that is not installed. Check /etc/tdnf/tdnf.conf
-
-Fixes: Bug 24144834: TDNF does not support $basearch and $releasever in .repo files - distroverpkg error
-
-By default, determine $ basearch and $ releasever from this package:
-mariner_user@mariner-aks [ /etc/tdnf ]$ sudo tdnf --refresh list installed | grep mariner
-mariner-release.noarch                      1.0-1.cm1                @System
-
-Workaround: add the following to /etc/tdnf/tdnf.conf
-distroverpkg=mariner-release
+---
+ client/defines.h | 2 +-
+ common/config.h  | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)
 
 diff --git a/client/defines.h b/client/defines.h
-index 7cb16b8..5131040 100644
+index 602b249..59b058e 100644
 --- a/client/defines.h
 +++ b/client/defines.h
-@@ -121,7 +121,7 @@ typedef enum
+@@ -137,7 +137,7 @@ typedef enum
+    and configurable with "persistdir" at run time */
+ #define TDNF_DEFAULT_DB_LOCATION          HISTORY_DB_DIR
+ 
+-#define TDNF_DEFAULT_DISTROVERPKG         "system-release"
++#define TDNF_DEFAULT_DISTROVERPKG         "mariner-release"
```

Comment:
Btw. I feel now all of this could be avoided, if we just added `Provides: system-release = %{version}-%{release}` inside `mariner-release.spec` for the sake of compatibility. It looks like we already have `Provides: system-release(releasever)` there anyway, so if you have a minute, it might be worth checking, if we could completely drop this patch with that change. However, it's a P2 at this point.